src/sentry/incidents/subscription_processor.py

```diff
@@ -633,19 +634,65 @@ def trigger_resolve_threshold(
     def handle_trigger_actions(
         self, incident_triggers: List[IncidentTrigger], metric_value: float
     ) -> None:
-        actions = deduplicate_trigger_actions(triggers=deepcopy(incident_triggers))
+        actions = deduplicate_trigger_actions(triggers=deepcopy(self.triggers))
         # Grab the first trigger to get incident id (they are all the same)
         # All triggers should either be firing or resolving, so doesn't matter which we grab.
         incident_trigger = incident_triggers[0]
-        method = "fire" if incident_triggers[0].status == TriggerStatus.ACTIVE.value else "resolve"
+        method = "fire" if incident_trigger.status == TriggerStatus.ACTIVE.value else "resolve"
+        try:
+            incident = Incident.objects.get(id=incident_trigger.incident_id)
+        except Incident.DoesNotExist:
+            # TODO: increment metric??
+            return
+
+        incident_activities = IncidentActivity.objects.filter(incident=incident)
+        past_statuses = {int(i.value) for i in incident_activities if i.value is not None}
+
+        critical_actions = []
+        warning_actions = []
         for action in actions:
+            if action.alert_rule_trigger.label == CRITICAL_TRIGGER_LABEL:
+                critical_actions.append(action)
+            else:
+                warning_actions.append(action)
+
+        actions_to_fire = []
+        new_status = IncidentStatus.CLOSED.value
+
+        if method == "resolve":
```

Comment:
If I'm understanding correctly, this logic used to live in `src/sentry/incidents/tasks.py` and has been relocated + expanded here. Can you walk me through why we're rehoming this chunk?

Reply:
There's a slight difference between the old and new logic for this. In the other file, the logic was to determine whether we should change the message of a _single_ action. For example, in the old logic, if a critical trigger was resolving and a warning trigger exists, we would change the action to send warning status instead of resolved status. Now, we're not making any of those per-action changes, we want to determine _one_ status to send to *all* actions. That's why it makes sense to pull the logic up a level to here.

Reply:
Great explanation, thank you!